### DISCUSSION TITLE
Use the old Copilot CLI syntax in the new gh version

### DISCUSSION THREAD

**Discussion by NullDev** [community/community#86354] [npm] [answered]
### Select Topic Area

Question

### Body

I still had the old `github-copilot-cli` v0.1.21 installed where I could just do e.g. `?? list all files`.
This doesn't work anymore (`ERR_BAD_REQUEST`) so I wanted to switch to the official [gh cli](https://github.com/cli/cli) that apparently has the copilot cli now. 

I uninstalled the old one with `npm rm -g @githubnext/github-copilot-cli` and installed `gh` as per the docs. 

Now, for the old one I still had aliases in my .bashrc / .zshrc: 
```sh
eval "$(github-copilot-cli alias -- "$0")"
```

So I could do `?? query`.

Is there any official way to get the old syntax back without limiting any new features? 

Always typing `gh copilot suggest "foo"` seems really verbose and I really liked the intuitive syntax with `??`.

**Comment by johanrosenkilde** [comment, marked as answer]
You may add the following aliases into your `.bashrc` or `.zshrc`:
```bash
alias '??'='gh copilot suggest -t shell'
alias 'git?'='gh copilot suggest -t git'
alias 'explain'='gh copilot explain'
```

This allows calling e.g. `?? list all files` like with the Next technical preview Copilot CLI. The difference is that you still "accept" by getting the command into your clipboard, at which point you should exit the cli and paste in the command in the shell.

In an upcoming version of the gh version, it will be born with a feature more closely matching the Next technical preview's in terms of aliases for quick execution.

**Reply by NullDev** [reply]
Awesome, thanks!

**Reply by solrevdev** [reply]
Yeah thanks for the info 🙏

**Reply by huyz** [reply]
Don't forget ;)
```shell
alias 'gh?'='gh copilot suggest -t gh'
```

**Comment by rsmets** [comment]
I'm curious which new version the [gh cli](https://github.com/cli/cli/releases) has copilot support? I can not find mention of it anywhere in the docs. 

_I'd love to upgrade because my old `@githubnext/github-copilot-cli` cli setup is no longer working - always throwing 404s (even after upgrading and re-authing). https://github.com/orgs/community/discussions/80191#discussioncomment-8081884_

**Reply by rsmets** [reply]
Ah found the [install instructions](https://docs.github.com/en/copilot/github-copilot-in-the-cli/using-github-copilot-in-the-cli#installing-copilot-in-the-cli) for the copilot cli extension 👍🏼

**Comment by johanrosenkilde** [comment]
The Next GitHub Copilot CLI technical preview is up and running again!

It was an unfortunate oversight when we sunset a few other previews in December that our Copilot CLI also stopped working. We *will* sunset our Copilot CLI technical preview in the not too far future, once the gh cli extension matures sufficiently.

**Comment by marcelosalloum** [comment]
It started working for me again after I updated the copilot package:
```
yarn global add @githubnext/github-copilot-cli
```

Maybe they messed up with the old DNS address?

**Reply by NullDev** [reply]
Apparently they fixed it 11 hours ago
https://github.com/community/community/discussions/86354#discussioncomment-8107247

**Comment by stikoo** [comment]
I've been able to use the `?? list all files` type syntax for ages, but just recently I think I updated copilot with Brew, and now that syntax won't work. I'm now on v1.0.4 of copilot cli after adding the extension. Just wondered if I can reinstate the `??` syntax easily?

**Reply by solrevdev** [reply]
You can add some aliases see above https://github.com/orgs/community/discussions/86354#discussioncomment-8081865 but here they are again:

```bash
alias '??'='gh copilot suggest -t shell'
alias 'git?'='gh copilot suggest -t git'
alias 'explain'='gh copilot explain'
alias 'gh?'='gh copilot suggest -t gh'
```

**Reply by stikoo** [reply]
Thanks, I did try this, but didn't have much luck, my `.bashrc` looks like this...

```
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion

export PATH="$HOME/.yarn/bin:$HOME/.config/yarn/global/node_modules/.bin:$PATH"

eval "$(gh copilot alias -- bash)"

alias '??'='gh copilot suggest -t shell'
alias 'git?'='gh copilot suggest -t git'
alias 'explain'='gh copilot explain'
alias 'gh?'='gh copilot suggest -t gh'
```

When my bash prompt starts up I get a `bash: github-copilot-cli: command not found` warning, so I think I've maybe got some weird issues going on. Maybe the transition from the npm package copilot to the brew installed version is not playing nice.

Again, running the `?? list all files` command returns a `bash: js: command not found` error, maybe I've got an old npm alias hanging around still or something, can't remember where that might be though, any ideas?

**Reply by solrevdev** [reply]
Yes that sounds plausible why not try removing the old npm version first then installing via gh then setting up the aliases so:

Uninstall NPM version:
```bash
npm uninstall -g @githubnext/github-copilot-cli
```

Make sure you have latest gh
```bash
brew install gh
brew upgrade gh
```

Install the copilot cli within gh
```bash
gh auth login
gh extension install github/gh-copilot
gh extension upgrade gh-copilot
```

Then in `.zshrc` if you use zsh which I think macOS now defaults to

```bash
alias '??'='gh copilot suggest -t shell'
alias 'git?'='gh copilot suggest -t git'
alias 'explain'='gh copilot explain'
alias 'gh?'='gh copilot suggest -t gh'

eval "$(gh copilot alias -- zsh)"
```

However if you definitely use bash  then in your `.bashrc` file

```bash
alias '??'='gh copilot suggest -t shell'
alias 'git?'='gh copilot suggest -t git'
alias 'explain'='gh copilot explain'
alias 'gh?'='gh copilot suggest -t gh'

eval "$(gh copilot alias -- bash)"
```

**Reply by stikoo** [reply]
Thanks @solrevdev, after re-googling my bash error, someone recommended just globally reinstalling the @githubnext/github-copilot-cli package. When I did this, things starting working again. I think this is probably because I recently change Node version with NVM that the copilot package was not being found.

I must have an alias elsewhere that is handling my ?? terminal command as it defo isn't in my .bashrc (I'm on an old mac so still using bash). I think I would prefer to not use the node version, but can't remember where the heck this alias could be, so will stick it out for now with a view to use the above when I work it out 🤣

**Reply by solrevdev** [reply]
Ahh well at least it’s working!

The ?? Is being handled by the old node @githubnext/github-copilot-cl package would be my guess as when you move to the gh version the alias changes and ?? Is no longer a thing and so you have to recreate it as your own alias as above. 

Hope this helped!

**Comment by awladux** [comment]
The ?? Is being handled by the old node @githubnext/github-copilot-cl package would be my guess as when you move to the gh version the alias changes and ?? Is no longer a thing and so you have to recreate it as your own alias as above.
